compiler/src/symboltable.rs

```diff
@@ -1120,12 +1120,33 @@ impl SymbolTableBuilder {
             // Role already set..
             match role {
                 SymbolUsage::Global => {
-                    if !symbol.is_global() {
```

Comment:
let's keep this if statement on the top of these errors. it will decrease hot path cost.

Reply:
Ok! I agree with that. I sent v2 code.